### ISSUE TITLE
Fail to enforce EOL with one single empty line.

### ISSUE THREAD

**Issue by lijunle** [eslint/eslint#4140]
Fail to enforce EOL with one single empty line.

My configuration:

```json
{
  "rules": {
    "eol-last": 2,
    "no-multiple-empty-lines": [2, {"max": 1}]
  }
}
```

Related: https://github.com/eslint/eslint/pull/1913 https://github.com/eslint/eslint/issues/1460#issuecomment-63178505

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by lijunle** [comment]
> The version of ESLint you are using (run eslint -v)

v1.5.0

> What you did (the source code and ESLint configuration)

File `a.js`, notice two empty line at the line of the file:

```js
require('something');


```

File `.eslintrc`:

```json
{
  "rules": {
    "eol-last": 2,
    "no-multiple-empty-lines": [2, {"max": 1}]
  }
}
```

> The actual ESLint output complete with numbers

Nothing is output.

> What you expected to happen instead

Expect warning me that there is multiple empty lines at the end of the file.

**Comment by gyandeeps** [comment]
Where are the empty lines?

**Comment by lijunle** [comment]
@gyandeeps The markdown just eat my double empty line. Let me update the code snippet.

**Comment by gyandeeps** [comment]
Based on the description you are saying that `no-multiple-empty-lines` rule is not able to catch the empty lines but the heading of this issue is going in different direction. I am lil confused here.
Can you please clarify what you are trying to report here?

**Comment by lijunle** [comment]
Well, if you look on #1913 and https://github.com/eslint/eslint/issues/1460#issuecomment-63178505, that makes a breaking change - it will not enforce `eol-last` checks that there is exactly one new line at the end of the file. Besides, it says, we should use `eol-last` with `no-multiple-empty-lines` rule to enforce **there is exactly one new line at the end of the file**. However, I failed.

**Comment by nzakas** [comment]
I think the point is that `no-multiple-empty-lines` isn't picking up empty lines at the end of input.

**Comment by alberto** [comment]
I think this is the culprit: https://github.com/eslint/eslint/blob/1867addebe0212641c675f2a420025b49cff04c4/lib/rules/no-multiple-empty-lines.js#L57-L59

**Comment by alberto** [comment]
For clarification. no-multiple-empty-lines is picking up empty lines at the end of input, but it is not doing it correctly. There is an off-by-one bug caused by the lines I mentioned in the previous message.

**Comment by nzakas** [comment]
I think that exception was put it so it wouldn't flag empty lines at the end of the file, since there's a separate rule for that. We need to look at that relationship closely.

**Comment by alberto** [comment]
`eol-last` only ensures there is at least one empty line (see comments above, behaviour was relaxed). Currently, there is a test in `no-multiple-empty-lines` that considers a final empty line a valid test case:
```js
        {
            code: "// valid 5\nvar a = 5;\n",
            options: [{ max: 0 } ]
        },
```

I think if we keep that test passing we are safe here.

**Comment by alberto** [comment]
I pushed the bug and proposed fix to my fork, I hope it makes it more clear.

**Comment by lijunle** [comment]
I think [this approach](https://github.com/eslint/eslint/pull/4266) is better. Thanks all! :smiley:
